### DISCUSSION TITLE
ModuleNotFoundError: No module named 'pyspark'

### DISCUSSION THREAD

**Discussion by Pythoneryeah** [coder/code-server#6346] [Q&A] [answered]
1. Using jupyter/ Pyspark-Notebook as the image base, start code-server in docker mode, but why is it told that pyspark moudle does not exist when I execute spark program?
![image](https://github.com/coder/code-server/assets/37642828/5315fb0d-d3c3-42ef-854e-b8d8528761ee)

2. Run jupyter/pyspark-notebook in the form of docker alone, you can run spark programs in jupyter

**Comment by code-asher** [comment]
It sounds like an issue with the environment, probably code-server is not running in the Conda environment or whatever they use to install pyspark.  It sounds like they have an entrypoint that sets the environment but if you are running `code-server` it is probably bypassing that.  Maybe you just need to run `conda activate` before running code-server or something like that?

https://github.com/jupyter/docker-stacks/issues/1782#issuecomment-1241062902

**Reply by Pythoneryeah** [reply, marked as answer]
Thank you. I solved the problem. But still don't know why? Here's my solution:
1, the mkdir ~ /. Ipython/kernels/pyspark
2, vim ~ /. Ipython/kernels/pyspark/kernel. The json
3. Add the following:
{
"display_name": "pySpark",
"language": "python",
"argv": [
"/usr/local/anacond/bin/python3",
"-m",
"IPython.kernel",
"-f",
"{connection_file}"
].
"env": {
"SPARK_HOME": "/usr/local/spark",
"PYTHONPATH" : "/ usr/local/spark/python: / usr/local/spark/python/lib/py4j - 0.10.4 - SRC. Zip",
"PYTHONSTARTUP": "/usr/local/spark/python/pyspark/shell.py ",
"PYSPARK_SUBMIT_ARGS": "pyspark-shell"
}
}

![1689929782229](https://github.com/coder/code-server/assets/37642828/de5f0d65-87a7-430c-8bcf-436fff13a0a2)

**Reply by code-asher** [reply]
Nice!  One or all of those environment variables must be required to tell Python how to find pyspark.
